src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php

```diff
@@ -263,11 +263,26 @@ public function getContent()
         return $this->data['content'];
     }
 
+    public function getPrettyJSONContent()
+    {
+        return json_encode(json_decode($this->getContent()), JSON_PRETTY_PRINT);
+    }
+
     public function getContentType()
     {
         return $this->data['content_type'];
     }
 
+    public function getRequestContentType()
+    {
+        return $this->data['request_headers']['content-type'];
+    }
+
+    public function isRequestJSON()
+    {
+        return 1 === preg_match('{^application/[a-z]*?\+?json$}i', $this->getRequestContentType());
```

Comment:
matches `application/foojson` .. not sure it's intended. If not: `^application/(?:[a-z]+\+)?json$` is maybe better